diff --git a/cupy_setup_build.py b/cupy_setup_build.py
index e1f678e30..fb84e5332 100644
--- a/cupy_setup_build.py
+++ b/cupy_setup_build.py
@@ -838,7 +838,7 @@ class _UnixCCompiler(unixccompiler.UnixCCompiler):
 
             cuda_version = build.get_cuda_version()
             postargs = _nvcc_gencode_options(cuda_version) + [
-                '-O2', '--compiler-options="-fPIC"', '--std=c++11']
+                '-O2', '--compiler-options="-fPIC"', '--std=c++14']
             print('NVCC options:', postargs)
 
             return unixccompiler.UnixCCompiler._compile(