diff --git a/cupy/core/include/cupy/cub b/cupy/core/include/cupy/cub
index c3cceac11..0158fa19f 160000
--- a/cupy/core/include/cupy/cub
+++ b/cupy/core/include/cupy/cub
@@ -1 +1 @@
-Subproject commit c3cceac115c072fb63df1836ff46d8c60d9eb304
+Subproject commit 0158fa19f28619886232defd412433974af89611
diff --git a/cupy_setup_build.py b/cupy_setup_build.py
index 89fb70d3a..872129cb7 100644
--- a/cupy_setup_build.py
+++ b/cupy_setup_build.py
@@ -866,7 +866,7 @@ class _UnixCCompiler(unixccompiler.UnixCCompiler):
 
             cuda_version = build.get_cuda_version()
             postargs = _nvcc_gencode_options(cuda_version) + [
-                '-O2', '--compiler-options="-fPIC"', '--std=c++11']
+                '-O2', '--compiler-options="-fPIC"', '--std=c++14']
             print('NVCC options:', postargs)
 
             return unixccompiler.UnixCCompiler._compile(